packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -48,25 +52,25 @@ function styleFunctionSx(props) {
   let css = {};
 
   Object.keys(styles).forEach((styleKey) => {
-    if (typeof styles[styleKey] === 'object') {
+    const value = callIfFn(styles[styleKey], theme);
```

Comment:
This was wrong previously, if the value is a function we need to resolve it and do the rest of the logic.